lib/src/edwards25519/field_element.dart

```diff
@@ -0,0 +1,1038 @@
+import 'package:solana/src/edwards25519/extensions.dart';
```

Comment:
It's over 1,000 LOC without any hint of what's going on here. Don't be that cruel to someone who's going to read this code.

Besides that, it's not Dart code, it's `C` code written with Dart syntax.

Reply:
It's java code written in dart syntax. I am not cruel, I put a link to the original code. This is complex math and it's what it is. The tests pass and they also came from the same code base.